File: testdata/stringLiteral/expected/expectedStringLiteral.py
import pulumi
import pulumi_kubernetes as kubernetes

myapp_config_map = kubernetes.core.v1.ConfigMap("myappConfigMap",
    api_version="v1",
    kind="ConfigMap",
    metadata=kubernetes.meta.v1.ObjectMetaArgs(
        name="myapp",
    ),
    data={
        "key": "{\\\"uid\\\": \\\"$(datasource)\\\"}",
    })
myapp_var_config_map = kubernetes.core.v1.ConfigMap("myapp_varConfigMap",
    api_version="v1",
    kind="ConfigMap",
    metadata=kubernetes.meta.v1.ObjectMetaArgs(
        name="myapp-var",
    ),
    data={
        "key": "{\\\"uid\\\": \\\"${datasource}\\\"}",
    })
myapp_no_end_bracket_config_map = kubernetes.core.v1.ConfigMap("myapp_no_end_bracketConfigMap",
    api_version="v1",
    kind="ConfigMap",
    metadata=kubernetes.meta.v1.ObjectMetaArgs(
        name="myapp-no-end-bracket",
    ),
    data={
        "key": "{\\\"uid\\\": \\\"${datasource\\\"}",
    })
myapp_no_brackets_config_map = kubernetes.core.v1.ConfigMap("myapp_no_bracketsConfigMap",
    api_version="v1",
    kind="ConfigMap",
    metadata=kubernetes.meta.v1.ObjectMetaArgs(
        name="myapp-no-brackets",
    ),
    data={
        "key": "{\\\"uid\\\": \\\"$datasource\\\"",
    })

File: testdata/stringLiteral/expected/expectedStringLiteral.ts
import * as pulumi from "@pulumi/pulumi";
import * as kubernetes from "@pulumi/kubernetes";

const myappConfigMap = new kubernetes.core.v1.ConfigMap("myappConfigMap", {
    apiVersion: "v1",
    kind: "ConfigMap",
    metadata: {
        name: "myapp",
    },
    data: {
        key: "{\\\"uid\\\": \\\"$(datasource)\\\"}",
    },
});
const myapp_varConfigMap = new kubernetes.core.v1.ConfigMap("myapp_varConfigMap", {
    apiVersion: "v1",
    kind: "ConfigMap",
    metadata: {
        name: "myapp-var",
    },
    data: {
        key: "{\\\"uid\\\": \\\"${datasource}\\\"}",
    },
});
const myapp_no_end_bracketConfigMap = new kubernetes.core.v1.ConfigMap("myapp_no_end_bracketConfigMap", {
    apiVersion: "v1",
    kind: "ConfigMap",
    metadata: {
        name: "myapp-no-end-bracket",
    },
    data: {
        key: "{\\\"uid\\\": \\\"${datasource\\\"}",
    },
});
const myapp_no_bracketsConfigMap = new kubernetes.core.v1.ConfigMap("myapp_no_bracketsConfigMap", {
    apiVersion: "v1",
    kind: "ConfigMap",
    metadata: {
        name: "myapp-no-brackets",
    },
    data: {
        key: "{\\\"uid\\\": \\\"$datasource\\\"",
    },
});
